pydy/tests/test_joints.py

```diff
@@ -0,0 +1,36 @@
+#!/usr/bin/env python
+
+SYMPY_VERSION = sm.__version__
+
+from ..bodies import Body
+from ..joints import PinJoint
+from ..system import System
+
+
+class TestPinJoint():
+    def setup(self):
+        self.sys = System(method='joints')
+
+    def test_joints_parent_child_relationship(self):
+        body = Body('body')
+        pin_joint = PinJoint(self.ground, 0, body, sys.reference_frame.x * Symbol('l'))
+
+        assert self.sys.ground.child is None
+        assert body.parent is None
+
+        self.sys.add_joint(pin_joint)
+
+        assert self.sys.ground.child == body
+        assert body.parent == self.sys.ground
```

Comment:
So this is a new tree topology that somehow combines a Point and a ReferenceFrame?